### PR TITLE
Disable additional transforms for the PyTorch Autograd path

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Applying transforms on the trace after "forward backward split + transform_for_execution" is not a well-tested path and breaks the careful sorting of operations introduced inside the `split_forward_backward` function. In the future, we need to revisit the order of transformations and move execution transforms out of the `split_forward_backward` function. Today this PR is needed to restore distributed communication overlap with computation and unblock fixing TransformerEngine+DataParallel.

### CODE REVIEW

**Review by t-vi — APPROVED** [review]
Thank you @IvanYashchuk

**Comment by t-vi** [comment]
So I generally agree with the patch, but the CI has opinions.

File: thunder/__init__.py [hunk 1]
```diff
@@ -563,25 +563,38 @@ def get_computation_and_inputs(*args, **kwargs):
...
-            with langctxs.langctx(cd.langctx):
-                extraces = transform_for_execution(
-                    computation_trc,
-                    executors_list=cd.executors_list,
-                )
-            extrace = extraces[-1]
-            comp = extrace.python_callable()
+                    extraces = []
+                    check(
+                        not additional_transforms,
```
Comment (nikitaved) [inline thunder/__init__.py:576]: Ah, so, my suspicion was not something I have imagined while being too tired? :)

Comment (IvanYashchuk) [reply thunder/__init__.py:576]: I always confuse what do I need to pass to `check`.

**Comment by IvanYashchuk** [comment]
Notebooks fail with:
```
     18 cache_rec, i_, _ = thunder.compile_data(thunder_model).get_computation_and_inputs(x)
---> 19 computation_trace = cache_rec.computation_traces[0]

IndexError: list index out of range
```

**Comment by IvanYashchuk** [comment]
Next error in the notebooks:
```py
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
### DON'T TRY THIS AT HOME
computation_trace.bound_symbols[2].sym = cache_rec.computation_traces[0].bound_symbols[2].subsymbols[0].sym
if cache_rec.computation_traces[0].bound_symbols[3].subsymbols:
    computation_trace.bound_symbols[3] = cache_rec.computation_traces[0].bound_symbols[3].subsymbols[0]
computation_trace.bound_symbols[4].sym = cache_rec.computation_traces[0].bound_symbols[4].subsymbols[0].sym

wrap_as_highlighted_code(computation_trace)
```

**Pull request merged**